Controllers/AdminController.cs

```diff
@@ -35,46 +35,69 @@ public IActionResult CreateBook()
         {
 
             //TODO: Build a new CreateBookViewModel with a complete set of Writers from the database
-            var createVeiwModel = new CreateBookViewModel { Writers = _db.Writers };
-            return View("CreateBook", createVeiwModel); //TODO: pass the ViewModel onto the CreateBook View
+            var createViewModel = new CreateBookViewModel { Writers = _db.Writers };
+            return View("CreateBook", createViewModel); //TODO: pass the ViewModel onto the CreateBook View
         }
         [HttpPost]
         public IActionResult CreateBook(CreateBookViewModel bookVM)
         {
             //TODO: Build the Writer object using the parameter
-            //Book book = new Book() {
-            //    Name = bookVM.Name,
-            //    BookId = bookVM.BookId,
-            //    Title = bookVM.Title,
-            //    SKU = bookVM.SKU,
-            //    Price = bookVM.Price,
-            //    AuthorId = bookVM.Name,
-            //}
-
+            Writer writer;
+            if (bookVM.AuthorId > 0)
+            {
+                writer = _db.Writers.Where(w => w.Id.Equals(bookVM.AuthorId)).FirstOrDefault();
+            }
+            else
+            {
+                writer = new Writer();
+                writer.Name = bookVM.Name;
+                
+            }
             //TODO: Build the Book using the parameter data and your newly created author.
-
+            Book book = new Book()
+            {
+                Name = bookVM.Name,
+                BookId = bookVM.BookId,
+                Title = bookVM.Title,
+                SKU = bookVM.SKU,
+                Price = bookVM.Price,
+                AuthorId = bookVM.AuthorId,
```

Comment:
The Book Model has an Author not an AuthorId, so line 64 should read 
`Author = writer`